### PR TITLE
parse signlist locale agnostic

### PR DESCRIPTION
**Pull request by albfan**
In spanish the signlist output is:

```
línea=146 id=256 nombre=SignifyAdd prioridad=10
...
```

changing the parse for this lines, plugin works again

### CODE REVIEW

**Comment by jamessan** [comment]
Changing the regex will fail if the listed sign is part of a group, because that is displayed before the line number, when a group is set.

It would be better to use the sign related functions, rather than parsing the output of the commands.  However, those weren't available at the time and would require, I believe, changing the minimum supported versions of (n)vim.

**Comment by albfan** [comment]
Got it. I check for equivalent functions and it does exist https://vim-jp.org/vimdoc-en/sign.html#sign_getplaced()

I look for usages and found more places. The vim patch seems to be this:

https://github.com/vim/vim/commit/162b71479bd4dcdb3a2ef9198a1444f6f99e6843

patch 8.1.0614

let me rework it to check that patch to use functions

**Comment by albfan** [comment]
It should fall back correctly now. A fake sign to parse localized name can be a solution for previous versions, but probably not worth the pain.

**Comment by albfan** [comment]
Let me know if there's anything else to fix for this PR

File: autoload/sy/fold.vim [hunk 1]
```diff
@@ -88,11 +88,21 @@ endfunction
...
   let lines = []
-  for line in split(signlist, '\n')[2:]
-    call insert(lines, matchlist(line, '\v^\s+line\=(\d+)')[1], 0)
+
+  let has_sign_func = has('patch-8.1.614')
+  if has_sign_func
+    let signlist = sign_getplaced(b:sy.buffer)[0].signs
+  else
+    let signlist = split(sy#util#execute('sign place buffer='. b:sy.buffer), '\n')[2:]
+  endif
```
Comment (jamessan) [inline autoload/sy/fold.vim:98]: A better way to handle this is probably to define a wrapper in autoload/sy/util.vim.  Something like

```vim
if exists('*sign_getplaced')
  function! sy#util#get_signs(bufnr)
    return sign_getplaced(a:bufnr)
  endfunction
else
  function! sy#util#get_signs(bufnr)
    ... Parse ":sign place" output and return a similar data structure
  endfunction
endif
```

Then the places that call the wrapper code don't need to be complicated by the two approaches, and we only need to do the check once to define the function wrapper.

**Pull request merged**